org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/LambdaExpressionsTest.java

```diff
@@ -7399,6 +7399,51 @@ public void testBug529197_003() {
 			"SUCCESS.run"
 			);
 }
+/**
+ * Test that using many generic type arguments doesn't result in a compiler hang.
+ * See: https://github.com/eclipse-jdt/eclipse.jdt.core/issues/177
+ */
+public void testManyGenericsHangGh177() {
+	this.runConformTest(
+			new String[] {
+					"C0.java",
+					"""
+					public class C0
```

Comment:
Cool, yes this should come very close to the original.

I suggest to move it to XLargeTest2.

Ideally we'd add a performance-related assertion, but in the past such assertions caused some trouble, so I'm fine with adding it as a regular regression test, just in case and as a help for debugging should we see it hang again.